src/data/real/pi.lean

```diff
@@ -190,14 +190,15 @@ begin
   --     constructed from `u` tends to `0` at `+∞`
   let u := λ k : ℕ, (k:nnreal) ^ (-1 / (2 * (k:ℝ) + 1)),
   have H : tendsto (λ k : ℕ, (1:ℝ) - (u k) + (u k) ^ (2 * (k:ℝ) + 1)) at_top (𝓝 0),
-  { convert (((tendsto_rpow_div_mul_add (-1) 2 1 $ by norm_num).neg.const_add 1).add
+  { convert (((tendsto_rpow_div_mul_add (-1) 2 1 two_ne_zero.symm).neg.const_add 1).add
       tendsto_inv_at_top_zero).comp tendsto_coe_nat_at_top_at_top,
     { ext k,
       simp only [nnreal.coe_nat_cast, function.comp_app, nnreal.coe_rpow],
       rw [← rpow_mul (nat.cast_nonneg k) ((-1)/(2*(k:ℝ)+1)) (2*(k:ℝ)+1),
-          @div_mul_cancel _ _ _ (2*(k:ℝ)+1) (by { norm_cast, linarith }), rpow_neg_one k],
-      ring },
-    { simp } },
+         @div_mul_cancel _ _ _ (2*(k:ℝ)+1)
+            (by { norm_cast, simp only [nat.succ_ne_zero, not_false_iff]}), rpow_neg_one k,
+          tactic.ring.add_neg_eq_sub] },
```

Comment:
This one is ugly. Can't you use `sub_eq_add_neg.symm` or something like that?

Reply:
Yes, you are right: thanks!